<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Downloading and combining data from NCBI Genbank and BOLD</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Downloading and combining data from NCBI Genbank and BOLD</h1>



<p>This short vignette will show you how you can use the functions of the refdb package to download sequence data from two popular repositories: NCBI Genbank and BOLD. We will also cover how these data can be combined into a single reference database. For an introduction to the package refdb see the vignette <em>Introduction to the refdb package</em>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(refdb)</span></code></pre></div>
<div id="getting-data-from-ncbi-genbank" class="section level2">
<h2>Getting data from NCBI Genbank</h2>
<p>We start by downloading sequence data from NCBI. The refdb packages uses the rentrez package (Winter 2017) to interface with NCBI servers.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>silo_ncbi &lt;-<span class="st"> </span><span class="kw">refdb_import_NCBI</span>(<span class="st">&quot;Silo COI&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="co">#&gt; Downloading 72 sequences from NCBI...</span></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="co">#&gt; </span></span>
<span id="cb2-4"><a href="#cb2-4"></a>  <span class="op">|</span><span class="st">                                                                                                                                       </span></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="st">  </span><span class="er">|</span><span class="st">                                                                                                                                 </span><span class="er">|</span><span class="st">   </span><span class="dv">0</span>%</span>
<span id="cb2-6"><a href="#cb2-6"></a>  <span class="op">|</span><span class="st">                                                                                                                                       </span></span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="st">  </span><span class="er">|=================================================================================================================================|</span><span class="st"> </span><span class="dv">100</span>%</span></code></pre></div>
<p>Data already have a set of fields defined.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">refdb_get_fields</span>(silo_ncbi)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co">#&gt; source: source</span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">#&gt; id: id</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co">#&gt; taxonomy:</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">#&gt;   superkingdom: superkingdom</span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co">#&gt;   kingdom: kingdom</span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co">#&gt;   phylum: phylum</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co">#&gt;   subphylum: subphylum</span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="co">#&gt;   class: class</span></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="co">#&gt;   subclass: subclass</span></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co">#&gt;   infraclass: infraclass</span></span>
<span id="cb3-12"><a href="#cb3-12"></a><span class="co">#&gt;   order: order</span></span>
<span id="cb3-13"><a href="#cb3-13"></a><span class="co">#&gt;   suborder: suborder</span></span>
<span id="cb3-14"><a href="#cb3-14"></a><span class="co">#&gt;   infraorder: infraorder</span></span>
<span id="cb3-15"><a href="#cb3-15"></a><span class="co">#&gt;   superfamily: superfamily</span></span>
<span id="cb3-16"><a href="#cb3-16"></a><span class="co">#&gt;   family: family</span></span>
<span id="cb3-17"><a href="#cb3-17"></a><span class="co">#&gt;   genus: genus</span></span>
<span id="cb3-18"><a href="#cb3-18"></a><span class="co">#&gt;   species: species</span></span>
<span id="cb3-19"><a href="#cb3-19"></a><span class="co">#&gt; sequence: sequence</span></span>
<span id="cb3-20"><a href="#cb3-20"></a><span class="co">#&gt; marker: gene</span></span>
<span id="cb3-21"><a href="#cb3-21"></a><span class="co">#&gt; latitude: latitude</span></span>
<span id="cb3-22"><a href="#cb3-22"></a><span class="co">#&gt; longitude: longitude</span></span></code></pre></div>
</div>
<div id="getting-data-from-bold" class="section level2">
<h2>Getting data from BOLD</h2>
<p>Now let’s download data for another taxon from the BOLD database.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>goera_bold &lt;-<span class="st"> </span><span class="kw">refdb_import_BOLD</span>(<span class="dt">taxon =</span> <span class="st">&quot;Goera pilosa&quot;</span>, <span class="dt">ncbi_taxo =</span> <span class="ot">FALSE</span>)</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co">#&gt; Downloading 30 sequences from BOLD...</span></span></code></pre></div>
<p>You may have noticed that the search interface is a bit different. This is because here we rely on another package (bold, Chamberlain 2020) to download the data. You can check the manual of <code>refdb_import_BOLD</code> to see the different arguments available. For the purpose of this vignette we also have turned off the automatic conversion to the NCBI taxonomy.</p>
<p>Similarly to the NCBI data, data downloaded from BOLD have a set of fields defined automatically.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw">refdb_get_fields</span>(goera_bold)</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="co">#&gt; source: source</span></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co">#&gt; id: sequenceID</span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="co">#&gt; taxonomy:</span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="co">#&gt;   phylum: phylum_name</span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="co">#&gt;   class: class_name</span></span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="co">#&gt;   order: order_name</span></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">#&gt;   family: family_name</span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">#&gt;   subfamily: subfamily_name</span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co">#&gt;   genus: genus_name</span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">#&gt;   species: species_name</span></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co">#&gt; sequence: nucleotides</span></span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="co">#&gt; marker: markercode</span></span>
<span id="cb5-14"><a href="#cb5-14"></a><span class="co">#&gt; latitude: lat</span></span>
<span id="cb5-15"><a href="#cb5-15"></a><span class="co">#&gt; longitude: lon</span></span></code></pre></div>
</div>
<div id="merging-data-from-different-sources" class="section level2">
<h2>Merging data from different sources</h2>
<p>We can now use <code>refdb_merge</code> to merge the two databases. To make things clearer we will keep only the first three sequences of <code>goera_bold</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="co"># Extract the first three records of goera_bold</span></span>
<span id="cb6-2"><a href="#cb6-2"></a>goera_bold &lt;-<span class="st"> </span>goera_bold[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, ]</span>
<span id="cb6-3"><a href="#cb6-3"></a></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co"># Merging goera_bold and silo_ncbi into one database</span></span>
<span id="cb6-5"><a href="#cb6-5"></a>bold_ncbi &lt;-<span class="st"> </span><span class="kw">refdb_merge</span>(goera_bold, silo_ncbi)</span></code></pre></div>
<p>Note that you can merge more than two database with <code>refdb_merge</code>.</p>
<p>Now, let’s take a look at the columns <code>genus_name</code>, <code>species_name</code> and <code>nucleotides</code> of the merged database.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>bold_ncbi[, <span class="kw">c</span>(<span class="st">&quot;source&quot;</span>, <span class="st">&quot;genus_name&quot;</span>, <span class="st">&quot;species_name&quot;</span>, <span class="st">&quot;nucleotides&quot;</span>)]</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="co">#&gt; # A tibble: 75 x 4</span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&gt;    source genus_name species_name  nucleotides                                                                                             </span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;         &lt;DNA&gt;                                                                                                   </span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">#&gt;  1 BOLD   Goera      Goera pilosa  AACAATTTATTTTATTTTTGGTATTTGATCAGGAATAGTCGGAACGTCCCTAAGTATAATTATTCGAATTGAATTAGGAACAGCTAATTCTTTAATTAAAAAT…</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt;  2 BOLD   Goera      Goera pilosa  AACAATTTATTTTATTTTTGGTATTTGATCAGGAATAGTCGGAACATCCCTAAGTATAATTATTCGAATTGAATTAGGAACAGCTAATTCTTTAATTAAAAAT…</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt;  3 BOLD   Goera      Goera pilosa  AACAATTTATTTTATTTTTGGTATTTGATCAGGAATAGTCGGAACGTCCCTAAGTATAATTATTCGAATTGAATTAGGAACAGCTAATTCTTTAATTAAAAAT…</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt;  4 NCBI   Silo       Silo pallipes AACTATTTATTTTATTTTTGGAATTTGATCTGGAATAGTAGGAACTTCTCTAAGAATAATTATTCGAACTGAGCTAGGGACTGCTGAATCTTTAATTAAAAAT…</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt;  5 NCBI   Silo       Silo pallipes AACTATTTATTTTATTTTTGGAATTTGATCTGGAATAGTAGGAACTTCTCTAAGAATAATTATTCGAACTGAGCTAGGGACTGCTGAATCTTTAATTAAAAAT…</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#&gt;  6 NCBI   Silo       Silo pallipes AACTATTTATTTTATTTTTGGAATTTGATCTGGAATAGTAGGAACTTCTCTAAGAATAATTATTCGAACTGAGCTAGGGACTGCTGAATCTTTAATTAAAAAT…</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="co">#&gt;  7 NCBI   Silo       Silo pallipes AACTATTTATTTTATTTTTGGAATTTGATCTGGAATAGTAGGAACTTCTCTAAGAATAATTATTCGAACTGAGCTAGGGACTGCTGAATCTTTAATTAAAAAT…</span></span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="co">#&gt;  8 NCBI   Silo       Silo pallipes AACTATTTATTTTATTTTTGGAATTTGATCTGGAATAGTAGGAACTTCTCTAAGAATAATTATTCGAACTGAGCTAGGGACTGCTGAATCTTTAATTAAAAAT…</span></span>
<span id="cb7-13"><a href="#cb7-13"></a><span class="co">#&gt;  9 NCBI   Silo       Silo pallipes AACTATTTATTTTATTTTTGGAATTTGATCTGGAATAGTAGGAACTTCTCTAAGAATAATTATTCGAACTGAGCTAGGGACTGCTGAATCTTTAATTAAAAAT…</span></span>
<span id="cb7-14"><a href="#cb7-14"></a><span class="co">#&gt; 10 NCBI   Silo       Silo pallipes AACTATTTATTTTATTTTTGGAATTTGATCTGGAATAGTAGGAACTTCTCTAAGAATAATTATTCGAACTGAGCTAGGGACTGCTGAATCTTTAATTAAAAAT…</span></span>
<span id="cb7-15"><a href="#cb7-15"></a><span class="co">#&gt; # … with 65 more rows</span></span></code></pre></div>
<p>Despite those columns having different names in the two databases, they were merged successfully because they were associated to the same field. The names used in the merged database are inherited from the database used as first argument in the merge functions (here <code>goera_bold</code>).</p>
<p>By default only the columns that are associated to a field in one of the two databases are kept. Of course, if there is no equivalent in the other database, NAs are produced. See the columns <code>subfamily_name</code> (subfamily is a taxonomic rank which exist only BOLD):</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>bold_ncbi[, <span class="kw">c</span>(<span class="st">&quot;source&quot;</span>, <span class="st">&quot;subfamily_name&quot;</span>)]</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="co">#&gt; # A tibble: 75 x 2</span></span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="co">#&gt;    source subfamily_name</span></span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;         </span></span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co">#&gt;  1 BOLD   Goerinae      </span></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="co">#&gt;  2 BOLD   Goerinae      </span></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="co">#&gt;  3 BOLD   Goerinae      </span></span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="co">#&gt;  4 NCBI   &lt;NA&gt;          </span></span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="co">#&gt;  5 NCBI   &lt;NA&gt;          </span></span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="co">#&gt;  6 NCBI   &lt;NA&gt;          </span></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="co">#&gt;  7 NCBI   &lt;NA&gt;          </span></span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="co">#&gt;  8 NCBI   &lt;NA&gt;          </span></span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="co">#&gt;  9 NCBI   &lt;NA&gt;          </span></span>
<span id="cb8-14"><a href="#cb8-14"></a><span class="co">#&gt; 10 NCBI   &lt;NA&gt;          </span></span>
<span id="cb8-15"><a href="#cb8-15"></a><span class="co">#&gt; # … with 65 more rows</span></span></code></pre></div>
<p>Alternatively, we can request the function <code>refdb_merge</code> to return only the fields shared by all the reference databases.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>bold_ncbi &lt;-<span class="st"> </span><span class="kw">refdb_merge</span>(goera_bold, silo_ncbi, <span class="dt">keep =</span> <span class="st">&quot;fields_shared&quot;</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="kw">colnames</span>(bold_ncbi)</span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="co">#&gt;  [1] &quot;source&quot;       &quot;sequenceID&quot;   &quot;phylum_name&quot;  &quot;class_name&quot;   &quot;order_name&quot;   &quot;family_name&quot;  &quot;genus_name&quot;   &quot;species_name&quot; &quot;nucleotides&quot; </span></span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="co">#&gt; [10] &quot;markercode&quot;   &quot;lat&quot;          &quot;lon&quot;</span></span></code></pre></div>
<p>NCBI and BOLD share several taxonomic ranks (phylum, class, order, family, genus, and species) that are a good basis for sequence taxonomic classification. But we may also use the default behavior of <code>refdb_import_BOLD</code> to force BOLD data to conform to the NCBI taxonomic system. So let’s download these data again.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>goera_bold &lt;-<span class="st"> </span><span class="kw">refdb_import_BOLD</span>(<span class="dt">taxon =</span> <span class="st">&quot;Goera pilosa&quot;</span>, <span class="dt">ncbi_taxo =</span> <span class="ot">TRUE</span>)</span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="co">#&gt; Downloading 30 sequences from BOLD...</span></span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="co">#&gt; </span></span>
<span id="cb10-4"><a href="#cb10-4"></a>Processing<span class="op">:</span><span class="st"> </span>Goera</span>
<span id="cb10-5"><a href="#cb10-5"></a>goera_bold &lt;-<span class="st"> </span>goera_bold[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, ]</span>
<span id="cb10-6"><a href="#cb10-6"></a><span class="kw">refdb_get_fields</span>(silo_ncbi)</span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="co">#&gt; source: source</span></span>
<span id="cb10-8"><a href="#cb10-8"></a><span class="co">#&gt; id: id</span></span>
<span id="cb10-9"><a href="#cb10-9"></a><span class="co">#&gt; taxonomy:</span></span>
<span id="cb10-10"><a href="#cb10-10"></a><span class="co">#&gt;   superkingdom: superkingdom</span></span>
<span id="cb10-11"><a href="#cb10-11"></a><span class="co">#&gt;   kingdom: kingdom</span></span>
<span id="cb10-12"><a href="#cb10-12"></a><span class="co">#&gt;   phylum: phylum</span></span>
<span id="cb10-13"><a href="#cb10-13"></a><span class="co">#&gt;   subphylum: subphylum</span></span>
<span id="cb10-14"><a href="#cb10-14"></a><span class="co">#&gt;   class: class</span></span>
<span id="cb10-15"><a href="#cb10-15"></a><span class="co">#&gt;   subclass: subclass</span></span>
<span id="cb10-16"><a href="#cb10-16"></a><span class="co">#&gt;   infraclass: infraclass</span></span>
<span id="cb10-17"><a href="#cb10-17"></a><span class="co">#&gt;   order: order</span></span>
<span id="cb10-18"><a href="#cb10-18"></a><span class="co">#&gt;   suborder: suborder</span></span>
<span id="cb10-19"><a href="#cb10-19"></a><span class="co">#&gt;   infraorder: infraorder</span></span>
<span id="cb10-20"><a href="#cb10-20"></a><span class="co">#&gt;   superfamily: superfamily</span></span>
<span id="cb10-21"><a href="#cb10-21"></a><span class="co">#&gt;   family: family</span></span>
<span id="cb10-22"><a href="#cb10-22"></a><span class="co">#&gt;   genus: genus</span></span>
<span id="cb10-23"><a href="#cb10-23"></a><span class="co">#&gt;   species: species</span></span>
<span id="cb10-24"><a href="#cb10-24"></a><span class="co">#&gt; sequence: sequence</span></span>
<span id="cb10-25"><a href="#cb10-25"></a><span class="co">#&gt; marker: gene</span></span>
<span id="cb10-26"><a href="#cb10-26"></a><span class="co">#&gt; latitude: latitude</span></span>
<span id="cb10-27"><a href="#cb10-27"></a><span class="co">#&gt; longitude: longitude</span></span>
<span id="cb10-28"><a href="#cb10-28"></a><span class="kw">refdb_get_fields</span>(goera_bold)</span>
<span id="cb10-29"><a href="#cb10-29"></a><span class="co">#&gt; source: source</span></span>
<span id="cb10-30"><a href="#cb10-30"></a><span class="co">#&gt; id: sequenceID</span></span>
<span id="cb10-31"><a href="#cb10-31"></a><span class="co">#&gt; taxonomy:</span></span>
<span id="cb10-32"><a href="#cb10-32"></a><span class="co">#&gt;   superkingdom: superkingdom</span></span>
<span id="cb10-33"><a href="#cb10-33"></a><span class="co">#&gt;   kingdom: kingdom</span></span>
<span id="cb10-34"><a href="#cb10-34"></a><span class="co">#&gt;   phylum: phylum</span></span>
<span id="cb10-35"><a href="#cb10-35"></a><span class="co">#&gt;   subphylum: subphylum</span></span>
<span id="cb10-36"><a href="#cb10-36"></a><span class="co">#&gt;   class: class</span></span>
<span id="cb10-37"><a href="#cb10-37"></a><span class="co">#&gt;   subclass: subclass</span></span>
<span id="cb10-38"><a href="#cb10-38"></a><span class="co">#&gt;   infraclass: infraclass</span></span>
<span id="cb10-39"><a href="#cb10-39"></a><span class="co">#&gt;   order: order</span></span>
<span id="cb10-40"><a href="#cb10-40"></a><span class="co">#&gt;   suborder: suborder</span></span>
<span id="cb10-41"><a href="#cb10-41"></a><span class="co">#&gt;   infraorder: infraorder</span></span>
<span id="cb10-42"><a href="#cb10-42"></a><span class="co">#&gt;   superfamily: superfamily</span></span>
<span id="cb10-43"><a href="#cb10-43"></a><span class="co">#&gt;   family: family</span></span>
<span id="cb10-44"><a href="#cb10-44"></a><span class="co">#&gt;   genus: genus</span></span>
<span id="cb10-45"><a href="#cb10-45"></a><span class="co">#&gt;   species: species_name</span></span>
<span id="cb10-46"><a href="#cb10-46"></a><span class="co">#&gt; sequence: nucleotides</span></span>
<span id="cb10-47"><a href="#cb10-47"></a><span class="co">#&gt; marker: markercode</span></span>
<span id="cb10-48"><a href="#cb10-48"></a><span class="co">#&gt; latitude: lat</span></span>
<span id="cb10-49"><a href="#cb10-49"></a><span class="co">#&gt; longitude: lon</span></span></code></pre></div>
<p>Now we can observe many more matching taxonomic fields between the two database, which makes the merge operation much more straightforward. Note that you can operate NCBI taxonomic conversion with data from other sources using the function <code>refdb_set_ncbitax</code>.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Winter, D. J. (2017) rentrez: an R package for the NCBI eUtils API The R Journal 9(2):520-526</p>
<p>Chamberlain, S. (2020). bold: Interface to Bold Systems API. R package version 1.1.0. <a href="https://CRAN.R-project.org/package=bold" class="uri">https://CRAN.R-project.org/package=bold</a></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
